BioCro/DARPA/regional_runs_inputs/convert_weather_stlew.Rmd

```diff
@@ -0,0 +1,150 @@
+                            RH < 1 ~ RH))
+    biocro_met_location <- biocro_met %>% 
+      mutate(latitude = latlon$lat[point], 
+             longitude = latlon$lon[point])
+    biocro_met_locations <- rbind(biocro_met_locations, biocro_met_location)
+    biocro_met_path <- paste0("biocro_met_stlew/stlew-", 
+                              latlon$lat[point], "-", latlon$lon[point], 
+                              ".2010.csv")
+    write.csv(biocro_met, biocro_met_path, row.names = FALSE)
+  }
+  write.csv(biocro_met_locations, "biocro_met_stlew/biocro_met_stlew.csv", 
+            row.names = FALSE)
+}
+all_met <- read.csv("biocro_met_stlew/biocro_met_stlew.csv")
+```
+
+## Plots of weather data
+
+These are various plots to explore what is available in the weather data and the data itself. First modified date and year columns are added. 
+
+```{r}
+all_met_plot <- all_met %>% 
+  filter(year == 2008) %>% 
+  tidyr::unite(latitudelongitude, latitude:longitude) %>% 
+  mutate(date = as.Date(paste0(year, "-01-01")) - 1 + doy, 
+         time = case_when(
+           hour < 10 ~ paste0("0", hour, ":00"), 
+           hour >= 10 ~ paste0(hour, ":00")
+         ), 
+         datetime_gmt = as.POSIXct(paste(date, time), tz = "GMT"), 
+         datetime = as.POSIXct(format(datetime_gmt, tz = "America/Chicago")), 
+         date = as.Date(datetime, tz = "America/Chicago"), 
+         year_plot = format(date, "%Y"), 
+         latitudelongitude = as.factor(latitudelongitude)) 
+```
+
+Plot of one year of hourly weather variables. 
+
+```{r}
+all_met_plot_year <- all_met_plot %>% 
+  filter(year_plot == 2008 & latitudelongitude == "39.125_-101.375") %>%
+  select(datetime, latitudelongitude, solar:precip) %>% 
+  tidyr::gather(weather_var, weather_value, solar:precip)
+  
+ggplot(all_met_plot_year, aes(x = datetime, y = weather_value, color = latitudelongitude)) +
+  geom_line() +
+  facet_wrap(~weather_var, scales = "free_y")
+```
+
+Plot of one day of hourly weather variables. 
+
+```{r}
+all_met_plot_day <- all_met_plot %>% 
+  filter(date == as.Date("2008-01-01")) %>% 
+  select(datetime, latitudelongitude, solar:precip) %>% 
+  tidyr::gather(weather_var, weather_value, solar:precip)
+
+ggplot(all_met_plot_day, aes(x = datetime, y = weather_value, color = latitudelongitude)) +
+  geom_line() +
+  facet_wrap(~weather_var, scales = "free_y") +
+  theme(legend.position = "none")
+```
+
+Spatial plot of weather data location for temperature in 2020 at noon during May. 
+
+```{r}
+overall_map <- map_data("world") %>% 
+  filter(region == "USA" & is.na(subregion))
+
+spatial_met_plot <- all_met %>% 
+  filter(year == 2010, doy == 150, hour == 12)
+
+ggplot() +
+  geom_polygon(data = overall_map, aes(x = long, y = lat, group = group), 
+               fill = "white", color = "black") +
+  geom_raster(data = spatial_met_plot, aes(x = longitude, y = latitude, fill = Temp)) +
+  coord_quickmap()
+```
+
+## Subset ecoregions
```

Comment:
At some point while doing these regional runs, I think back when they extended across China and Russia, we wanted to display the ecological diversity across which the runs went. So that's why the ecoregions stuff is there. I've been kind of neglecting is for a while though because it doesn't seem that useful or informative.